### ISSUE TITLE
Microsoft.CodeAnalysis.BannedApiAnalyzers warn on parameter if used

### ISSUE THREAD

**Issue by michaelmairegger** [dotnet/roslyn#79289]
**Brief description:**

I have a BannedSymbols.txt file with the following content

    N:AutoMapper;Consider removing AutoMapper from your project. It will be commercial in v15

and expect that BannedApiAnalyzers will report any usage. Unfotunately it will report only when I call methods on the banned type, not the usage itself. E.g. if the banned type is used as constructor parameter or field then nothing will be reported.

**Languages applicable:**

C# (not tried in VB)

**Code example that the analyzer should report:**

   ```csharp
using AutoMapper;
public class Test
{
    private readonly IMapper _mapper; // <-- should warn here
    
    public Test(IMapper mapper) // <-- should warn here
    {
        _mapper = mapper;
    }
    
    public void T()
    {
        _mapper.Map<string>(5); // here correclty warned
    }
}
```
